pallets/pallet-migration/src/benchmarking.rs

```diff
@@ -0,0 +1,120 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2023 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+use frame_benchmarking::{account, benchmarks};
+use frame_support::{
+	traits::{
+		fungible::{Inspect, InspectFreeze, InspectHold, Mutate},
+		Get, LockableCurrency, ReservableCurrency, WithdrawReasons,
+	},
+	BoundedVec,
+};
+use frame_system::RawOrigin;
+use sp_runtime::{traits::IdentifyAccount, AccountId32, MultiSigner, SaturatedConversion};
+use sp_std::vec;
+
+use did::{
+	benchmarking::get_ed25519_public_authentication_key, did_details::DidVerificationKey,
+	mock_utils::generate_base_did_details, DidIdentifierOf,
+};
+use parachain_staking::migrations::STAKING_ID;
+use runtime_common::constants::KILT;
+
+use crate::*;
+
+const SEED: u32 = 0;
+type AccountIdOf<T> = <T as frame_system::Config>::AccountId;
+
+benchmarks! {
+	where_clause {
+		where
+		<T as attestation::Config>::Currency: ReservableCurrency<AccountIdOf<T>>,
+		<T as delegation::Config>::Currency: ReservableCurrency<AccountIdOf<T>>,
+		<T as did::Config>::Currency: ReservableCurrency<AccountIdOf<T>>,
+		<T as did::Config>::Currency: Mutate<AccountIdOf<T>>,
+		<T as did::Config>::Currency: Inspect<AccountIdOf<T>>,
+		<T as pallet_did_lookup::Config>::Currency: ReservableCurrency<AccountIdOf<T>>,
+		<T as pallet_web3_names::Config>::Currency: ReservableCurrency<AccountIdOf<T>>,
+		<T as parachain_staking::Config>::Currency: ReservableCurrency<AccountIdOf<T>>,
+		<T as parachain_staking::Config>::Currency: LockableCurrency<AccountIdOf<T>>,
+		<T as public_credentials::Config>::Currency: ReservableCurrency<AccountIdOf<T>>,
+		<T as did::Config>::DidIdentifier: From<AccountId32>,
+		<T as frame_system::Config>::AccountId: From<AccountId32>,
+	}
+	general_weight {
```

Comment:
Wouldn't it be easier to use as benchmarking variables the max value for each pallet? E.g. a tuple where each element is the limit for the respective pallet. Then weight computation becomes easier, as it is just picking the right value for the provided length in the parameter. Also, there is no need to have a special weight for staking, but it would be computed as part of the benchmark when the value of the staking limit is > 0.

TL;DR: Benchmarking should be performed over the max limit for each pallet as specified in the `requested_migrations` extrinsic. This requires some more complex mock setup to make sure enough accounts can be migrated, but results in more accurate and easier to compute weights.

Reply:
I updated the benchmark logic: [837e465](https://github.com/KILTprotocol/kilt-node/pull/540/commits/837e465995c18d6cd5184dacd51ffbec946664f0)

Reply:
I planned to take the calculated weights and multiply them by the sum of the requested migrations.